src/lib/token_list_gen.ts

```diff
@@ -67,87 +77,161 @@ export const generateTokenList = async (
     timestamp: new Date().toISOString(),
     version,
     tokens: tokenList,
-    logoURI: mainLogoUri // todo: handle undefined
+    logoURI: mainLogoUri
   };
   const res = validateTokenList(arbTokenList);
   if(!res){
-    console.log(arbTokenList);    
-    throw new Error("New token list invalid!")
+    console.log("Token list invalid — let's try to see why:");
+    while(arbTokenList.tokens.length > 0){
+      const candidateToken = arbTokenList.tokens.pop()
+      const res = validateTokenList(arbTokenList);
+      if (res){
+        console.log('This token is a problem:', candidateToken);
+        throw new Error('Invalid token list')
+
+      }
+    }
+    console.log(arbTokenList);
+    throw new Error('Invalid token list (not sure why)')
+    
   }
-  console.log(`Generated list with ${arbTokenList.tokens.length} tokens`);
+    //
+ console.log(`Generated list with ${arbTokenList.tokens.length} tokens`);
+ console.log('version:', version);
+ 
   
   return arbTokenList;
 };
 
 export const arbifyL1List = async (pathOrUrl: string) => {
   const l1TokenList = await getTokenListObj(pathOrUrl);
+  const path = process.env.PWD +
+  '/src/ArbTokenLists/' +
+  listNameToFileName(l1TokenList.name);
+  let prevArbTokenList: ArbTokenList | undefined; 
+
+  if(existsSync(path)){
+    const data = readFileSync(path)
+    console.log('Prev version of Arb List found');
+    
+    prevArbTokenList =  JSON.parse(data.toString()) as ArbTokenList
+  } 
 
   const l1Addresses = l1TokenList.tokens.map((token) =>
     token.address.toLowerCase()
   );
 
-  const newList = await generateTokenList(l1Addresses, l1TokenList.name, l1TokenList.logoURI);
-  const path =
-    process.env.PWD +
-    '/src/ArbTokenLists/' +
-    listNameToFileName(l1TokenList.name);
+  const newList = await generateTokenList(l1Addresses, l1TokenList.name, l1TokenList.logoURI, prevArbTokenList);
 
   writeFileSync(path, JSON.stringify(newList));
+  console.log('Token list generated at', path );
+  
 };
 
-export const updateArbifiedList = async (path: string) => {
-  const data = readFileSync(path);
-  const tokenList = JSON.parse(data.toString()) as ArbTokenList;
-  // const tokenList  = getTokenListObj(path)
-  // TODO
-
-  const l1Addresses = tokenList.tokens
-    .map((token) => token.extensions.l1Address)
-    .filter((x): x is string => !!x);
-  const newList = await generateTokenList(l1Addresses, tokenList.name, tokenList.logoURI, tokenList.version);
-  const versionBump = minVersionBump(tokenList.tokens, newList.tokens)
-  const newVersion = nextVersion(tokenList.version, versionBump)  
-  const newListWithNewVersion = {...newList, ...{version: newVersion}} as ArbTokenList
-  writeFileSync(path, JSON.stringify(newListWithNewVersion));
+export const updateArbifiedList = async (pathOrUrl: string) => {
+  // @ts-ignore
+  const arbTokenList:ArbTokenList = await getTokenListObj(pathOrUrl);
+  const path = process.env.PWD +
+  '/src/ArbTokenLists/' +
```

Comment:
this script fails for me because I don't have a ArbTokenLists folder
```suggestion
  '/src/' +
```